### ISSUE TITLE
NullReferenceException in WebResponseFactory.CreateResponse

### ISSUE THREAD

**Issue by lorenzotandin** [darrylwhitmore/NScrape#24]
When the response does not contain the header `Content-Type`, the following code throws a `NullReferenceException`:
```csharp
var url = "https://domain.com/imageWithNoContent-Type.jpg"
var response = new WebClient().SendRequest(new Uri(url));
```
StackTrace for the exception looks like this:
```
   at NScrape.WebResponseFactory.<>c__DisplayClass12_0.<CreateResponse>b__0(String k)
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at NScrape.WebResponseFactory.CreateResponse(HttpWebResponse webResponse)
   at NScrape.WebClient.SendRequest(WebRequest webRequest)
   at NScrape.WebClient.SendRequest(Uri destination)
```
Likely cause of the exception: in `WebResponseFactory`, at line 171, the method `CreateResponse` is using `contentType` retrieved from `webResponse.Headers[CommonHeaders.ContentType]` without first checking if it's not null.

**Comment by darrylwhitmore** [comment]
Hi Lorenzo, thanks for reporting this. That URL is now returning a 404; do you have another example URL that I can try? 

If not, I did simulate a missing **Content-Type** header in `NScrape.WebResonseFactory.CreateResponse()` in the debugger and got the same stack trace as you, so I can go ahead and make a fix and test it out using the same method in the debugger.

**Comment by lorenzotandin** [comment]
Hi Darryl,

that was just a fake url I came up with for example purposes.
I didn't know `domain.com` is an actual website :smiley:
I'm sorry, I cannot share the actual url I was having trouble with.

I think a reasonable fix might be inserting the following code in `WebResponseFactory.CreateResponse()`, at line 168, right after `contentType` is set:
```
if (contentType == null)
    return new UnsupportedWebResponse(webResponse.ResponseUri, contentType);
```

**Comment by darrylwhitmore** [comment]
Hi Lorenzo, doh, I think I should have probably figured that out 😄. Ok, I will put in a fix and just test in the debugger. Thanks again!

**Comment by darrylwhitmore** [comment]
Fixed in commit ca4de3a.
